### PR TITLE
Tests for Creating and opening mutex, semaphore and waithandles with long names

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/corefx/issues/3740
Related Coreclr PR https://github.com/dotnet/coreclr/pull/18402

we're letting the API kick back names that are too long rather than pre-validating them. 
This PR includes Tests for long names and removes existing tests which expects the exception to be thrown in these cases.
This squashed commit will need to be added whenever maestro brings the new version of the coreclr with the related changes

### CODE REVIEW

File: src/System.Threading/tests/SemaphoreTests.cs [hunk 1]
```diff
@@ -74,15 +73,6 @@ public void Ctor_InvalidArguments()
             AssertExtensions.Throws<ArgumentException>(null, () => new Semaphore(2, 1, "CtorSemaphoreTest", out createdNew));
         }
 
-        [PlatformSpecific(TestPlatforms.Windows)]  // named semaphores aren't supported on Unix
-        [Fact]
-        public void Ctor_InvalidNames()
```
Comment (jkotas) [inline src/System.Threading/tests/SemaphoreTests.cs]: I would guess that there is still limit on the name in the OS. Could you please try what happens for 10MB name?

File: src/System.Threading/tests/EventWaitHandleTests.cs [hunk 2]
```diff
@@ -224,5 +227,10 @@ private static int PingPong_OtherProcess(string modeName, string inboundName, st
             return SuccessExitCode;
         }
 
+        public static TheoryData<string> GetValidNames => new TheoryData<string>()
+        {
+            { Guid.NewGuid().ToString("N") },
+            { Guid.NewGuid().ToString("N") + new string('a', 1000) }
```
Comment (jkotas) [inline src/System.Threading/tests/EventWaitHandleTests.cs]: Does this case need to be disabled for .NET Framework?

File: src/System.Threading/tests/SemaphoreTests.cs (hunk 1 above)
Comment (Anipik) [reply src/System.Threading/tests/SemaphoreTests.cs]: I tried with ```10,000,000 * 2 bytes = 20 megabytes``` and the test is still not failing

Comment (jkotas) [reply src/System.Threading/tests/SemaphoreTests.cs]: Ok, you are right. I have tried as well and there does not seem to be any hard-coded limit.

File: src/System.Threading/tests/EventWaitHandleTests.cs (hunk 2 above)
Comment (Anipik) [reply src/System.Threading/tests/EventWaitHandleTests.cs]: Yeah I will disable it. should I re-add the test for ```Invalid Name``` for just .NET Framework ?

**Comment by Anipik** [comment]
@jkotas can you take a final look at these changes so that when maestro updates the version of the coreclr, we can add this commit on top  of that

**Comment by Anipik** [comment]
merged through maestro

**Pull request closed without merging**